EventFilter/L1TRawToDigi/plugins/implementations_stage2/BMTFUnpackerOutput.cc

```diff
@@ -13,78 +13,76 @@ namespace l1t
       else
-	bxBlocks = block.getBxBlocks((unsigned int)6, false);//it returnes 6-32bit bxBlocks originated from the amc13 Block
-			
+        bxBlocks = block.getBxBlocks((unsigned int)6, false);//it returnes 6-32bit bxBlocks originated from the amc13 Block
+
       RegionalMuonCandBxCollection *res;
       if (isTriggeringAlgo)
-	res = static_cast<BMTFCollections*>(coll)->getBMTFMuons();
+        res = static_cast<BMTFCollections*>(coll)->getBMTFMuons();
       else
-	res = static_cast<BMTFCollections*>(coll)->getBMTF2Muons();
+        res = static_cast<BMTFCollections*>(coll)->getBMTF2Muons();
 
       //BxBlocks changed the format of the blocks
       int firstBX = 0, lastBX = 0;
       int nBX = 0;
       if (!bxBlocks.empty()) {
-	nBX = bxBlocks[0].header().getTotalBx();//how many BX included in the BxBlock before Suppression
-	getBXRange(nBX, firstBX, lastBX);
-	res->setBXRange(firstBX, lastBX);
+        nBX = bxBlocks[0].header().getTotalBx();//how many BX included in the BxBlock before Suppression
+        getBXRange(nBX, firstBX, lastBX);
+        res->setBXRange(firstBX, lastBX);
       }
       else {
-	res->setBXRange(-2, 2);
-	LogDebug("L1T") << "No BXs included in the given Block. Set the BXRange to be (-2,2).";
-	return true;
+        res->setBXRange(-2, 2);
+        LogDebug("L1T") << "No BXs included in the given Block. Set the BXRange to be (-2,2).";
+        return true;
       }
 
       LogDebug("L1T") << "nBX = " << nBX << " firstBX = " << firstBX << " lastBX = " << lastBX;
-			
+
       int processor = block.amc().getBoardID() - 1;
       if ( processor < 0 || processor > 11 )
-	{
-	  edm::LogInfo ("l1t:stage2::BMTFUnpackerOutput::unpack") << "Processor found out of range so it will be calculated by the old way";
-	  if ( block.amc().getAMCNumber()%2 != 0 )
-	    processor = block.amc().getAMCNumber()/2 ;
-	  else
-	    processor = 6 + (block.amc().getAMCNumber()/2 -1);
-	}
+      {
+        edm::LogInfo ("l1t:stage2::BMTFUnpackerOutput::unpack") << "Processor found out of range so it will be calculated by the old way";
+        if ( block.amc().getAMCNumber()%2 != 0 )
+        processor = block.amc().getAMCNumber()/2 ;
+        else
+        processor = 6 + (block.amc().getAMCNumber()/2 -1);
+      }
 
       for (auto bxBlock : bxBlocks) {
-	int ibx = bxBlock.header().getBx();
-
-	for (auto iw = 0; iw < 6; iw+=2) {
-	  uint32_t raw_first = bxBlock.payload()[iw];//payload[ip+(ibx+lastBX)*6];
-	  uint32_t raw_secnd = bxBlock.payload()[iw+1];//payload[ip+(ibx+lastBX)*6];
-	  if ( raw_first == 0 )
-	    {
-	      LogDebug("L1T") << "Raw data is zero";
-	      continue;
-	    }
-					
-	  RegionalMuonCand muCand;
-	  RegionalMuonRawDigiTranslator::fillRegionalMuonCand(muCand, raw_first, raw_secnd, processor, tftype::bmtf);
-
-	  if (muCand.hwQual() == 0)
-	    continue;
-
-	  muCand.setLink(48 + processor);	//the link corresponds to the uGMT input
```

Comment:
I noticed that it's only set in the BMTF unpacker (and not the GMT regional muon unpacker) and then saw that it's anyway done indirectly in fillRegionalMuonCand()[1] because setTFIdentifiers() seems to do it[2] (for reference, setLink() is at [3]. Can you have a look to make sure I didn't miss something?

[1] https://github.com/dinyar/cmssw/blob/788557afd732cfc0c4afa1c30d10f6e4c8aff99a/L1Trigger/L1TMuon/src/RegionalMuonRawDigiTranslator.cc#L88
[2] https://github.com/dinyar/cmssw/blob/788557afd732cfc0c4afa1c30d10f6e4c8aff99a/DataFormats/L1TMuon/src/RegionalMuonCand.cc#L18
[3] https://github.com/dinyar/cmssw/blob/788557afd732cfc0c4afa1c30d10f6e4c8aff99a/DataFormats/L1TMuon/interface/RegionalMuonCand.h#L81

Reply:
Looks redundant indeed.